### ISSUE TITLE
Issue with bulk update

### ISSUE THREAD

**Issue by odiea** [jekkos/opensourcepos#393]
I just added the first tax and this is what I get after submit.

A PHP Error was encountered

Severity: Warning
Message: Invalid argument supplied for foreach()
Filename: models/Item_taxes.php
Line Number: 37
Backtrace:
File: C:\xampp\htdocs\feature\application\models\Item_taxes.php
Line: 37
Function: _error_handler

File: C:\xampp\htdocs\feature\application\controllers\Items.php
Line: 501
Function: save_multiple

File: C:\xampp\htdocs\feature\index.php
Line: 292
Function: require_once
{"success":true,"message":"You have successfully updated the selected items"}

**Comment by daN4cat** [comment]
I assume it's vs 3.0 (bootstrap one), correct?

**Comment by odiea** [comment]
Sorry. Yes it was from my pull about 5 min ago from bootstrapUI.

**Comment by daN4cat** [comment]
Bulk update is flaw. It managed to reset the supplier from all the items in my test database (crap!)

**Comment by jekkos** [comment]
I'lll use this issue to track progress on bulk edit. Following points need to be checked

* supplier selection 
* update vat percentages 
* erorr described here

**Comment by daN4cat** [comment]
Supplier selection should have a "Do Nothing"

**Comment by odiea** [comment]
It appears that bulk update is still broken.

**Comment by daN4cat** [comment]
yes, we know.

**Comment by daN4cat** [comment]
See #381 for 2.4 behaviour.

**Comment by ais-one** [comment]
Hi, yes the behaviour is truely funky, I am trying to find out what is going on and hopefully fix this part, hopefully I can find out what is the problem. Even if the entry is "correct" after submit it shows the JSON result.

**Comment by daN4cat** [comment]
That's great! Thanks @ais-one

**Comment by ais-one** [comment]
Hi, I think I see what is causing the problem:

FILE: application\views\items\form_bulk.php

there was an error on Firebug about the with the category autocomplete:

Error: cannot call methods on autocomplete prior to initialization; attempted to call method 'http://127.0.0.1/opensourcepos/index.php/items/suggest_category'

This basically caused the form submit to have problem. Once I commented it out below, it worked ok.

```
$(document).ready(function()
{	
/*
    $("#category").autocomplete("<?php echo site_url('items/suggest_category');?>",`{max:100,minChars:0,delay:10});
   $("#category").result(function(event, data, formatted)
    {
    });
    $("#category").search();
*/
```

**Comment by daN4cat** [comment]
Yes, but we need to fix the category suggestion. Probably the controller method is not right or was removed?

**Comment by ais-one** [comment]
ok, let me check the suggest_category url

**Comment by daN4cat** [comment]
Hang on, the autocomplete is different now. You cannot pass the url like that. Check the item search in Items/manage. Also the controller needs to return json. I'm far from the code so I go by memory.

**Comment by ais-one** [comment]
Did the jQuery version change? 

It works after I change from

```
$("#category").autocomplete("<?php echo site_url('items/suggest_category');?>",{max:100,minChars:0,delay:10});
```

to

```
$("#category").autocomplete({source: "<?php echo site_url('items/suggest_category');?>",max:100,minChars:0,delay:10});
```

If it is ok, I will do a pull request of this but for master or bootstrapUI?

**Comment by ais-one** [comment]
You are right, there is alot more to fix, ok this make take alittle while more...

**Comment by ais-one** [comment]
The autocomplete is not playing well with the bootstrapUI.

 I did a test list with many categories. I could see the list, but it is behind the dialog box (its a list of programming languages)...

![list](https://cloud.githubusercontent.com/assets/4284574/13904992/0dac3d12-eeee-11e5-8dfc-32bcd1a9d74a.png)

**Comment by ais-one** [comment]
I had to add ui-front to a parent container of the category input

**Comment by ais-one** [comment]
ok, autocomplete, its working properly, shall I submit a pull request? Only 1 file changed, form_bulk.php

**Comment by daN4cat** [comment]
Yes, please submit a PR against feature/bootstrapUI branch. Thanks a lot for the help, well done!

**Comment by daN4cat** [comment]
BTW, outside the PHP error the bulk edit needs to be fixed in a couple of other points as written above.

**Comment by ais-one** [comment]
Pull request sent, this is for bulk, edit. I will check if there is more autocompletes in other places that need fixing later.

[https://github.com/jekkos/opensourcepos/pull/415]()

**Comment by jekkos** [comment]
Updating tax percentages in bulk update isn't really working right and probably can't be fix due to a flaw in the data model. This statement holds for 2.4 and 3.0.

I suggest to add some kind of warning in the confirm dialog shown after the submit is clicked. Also some weird things happen once the first field is left empty and only the seconnd is filled in.
If any other suggestion to fix this then post away.

**Comment by daN4cat** [comment]
The category suggestion works, but when you save the item with the new category nothing is saved/changed.

So the bulk edit is still quite broken.

**Comment by daN4cat** [comment]
Fixed bulk edit: save category, don't remove supplier is none is selected.
I also removed the annoying dialog box popping up say are you sure... it's always yes otherwise why have you pushed submit?

**Comment by ais-one** [comment]
@daN4cat 

I think the 'category' portion is one of the few fields ignored in the bulk edit (there is an in_array test), I thought it was a normal thing so I did not touch that portion

**Comment by daN4cat** [comment]
Nope, that was a mistake as it's not an array. Anyway sorted. 2.4 needs a fix to category and suppliers, do you want to look into that?

**Comment by daN4cat** [comment]
Applied category & supplier fix to 2.4 too.

**Comment by odiea** [comment]
A little late but bulk update is working great now.
